pkg/mtu/mtu.go

```diff
@@ -101,7 +105,7 @@ type Configuration struct {
 // specified, otherwise it will be automatically detected. if encapEnabled is
 // true, the MTU is adjusted to account for encapsulation overhead for all
 // routes involved in node to node communication.
-func NewConfiguration(authKeySize int, encryptEnabled bool, encapEnabled bool, wireguardEnabled bool, mtu int, mtuDetectIP net.IP) Configuration {
+func NewConfiguration(authKeySize int, encryptEnabled bool, encapEnabled bool, wireguardEnabled bool, hsIpcacheDSRenabled bool, mtu int, mtuDetectIP net.IP) Configuration {
```

Comment:
Drive by comment: this really ought to take a struct. All of the bools makes this unreadable at callsites.

Nit: capitalization of `hsIpcacheDSRenabled` seems wonky.

Reply:
No disagreement from my side, this has grown way out of proportion :grimacing: .